[10.0][ADD] account_multicurrency_revaluation_rate_type

i-vyshnevska opened the pull request:
functional from https://github.com/OCA/account-closing/pull/98 was moved to separate addon 
this module allow to choose  which currency type use when revaluation i executed
also some  comments was added to account_multicurrency_revaluation

blocked by https://github.com/OCA/account-closing/pull/102

Commit:
[ADD] account_multicurrency_revaluation_type: create module

Commit:
[IMP] account_multicurrency_revaluation: add todo

i-vyshnevska commented:
@grindtildeath can you give some attention  to this PR

sebalix commented on account_multicurrency_revaluation_type/tests/test_currency_revaluation.py:
```diff
@@ -0,0 +1,173 @@
+        move_ids = self.env['account.move.line'].browse(move_ids)
+        return move_ids
+
+    def wizard_execute(self, date, type):
```
rename `type` (Python keyword) to `rate_type`

sebalix commented on account_multicurrency_revaluation_type/tests/test_currency_revaluation.py:
```diff
@@ -0,0 +1,173 @@
+
+    def _get_move_ids(self, result):
+        move_ids = []
+        lines = result['domain'].split(',')
```
can't we use `safe_eval` to get the domain?

sebalix commented on account_multicurrency_revaluation_type/model/res_company.py:
```diff
@@ -0,0 +1,16 @@
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    rate_type = fields.Selection(
```
I would name this field `currency_rate_type` instead, easier to understand what we are talking about in the UI/database schema.

sebalix commented on account_multicurrency_revaluation_type/wizard/wizard_currency_revaluation.py:
```diff
@@ -0,0 +1,69 @@
+
+    _inherit = 'wizard.currency.revaluation'
+
+    # Todo get in count direct/indirect rate
```
Is this TODO still relevant? If it's optional, can it be added to the ROADMAP section?

i-vyshnevska replied:
yes, but it should be done in other module

sebalix commented on account_multicurrency_revaluation_type/wizard/wizard_currency_revaluation_view.xml:
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_account_config_settings" model="ir.ui.view">
```
`view_account_currency_revaluation_wizard` (`view_account_config_settings` already used in `views/res_config_view.xml`)

mpanarin commented on account_multicurrency_revaluation_type/__manifest__.py:
```diff
@@ -0,0 +1,22 @@
+        "account_multicurrency_revaluation",
+        "currency_monthly_rate",
+    ],
+    "demo": [
```
remove the empty block

mpanarin commented on account_multicurrency_revaluation_type/tests/test_currency_revaluation.py:
```diff
@@ -0,0 +1,173 @@
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestCurrencyRevaluationType, cls).setUpClass()
```
`cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))`

mpanarin commented on account_multicurrency_revaluation_type/tests/test_currency_revaluation.py:
```diff
@@ -0,0 +1,173 @@
+        cls.dec_31 = '%s-12-31' % cls.year
+
+        cls.company.revaluation_loss_account_id = \
+            cls.env.ref('account_multicurrency_revaluation.acc_reval_loss').id
```
why not use the ref method you initialized earlier? )

mpanarin commented on account_multicurrency_revaluation_type/wizard/wizard_currency_revaluation.py:
```diff
@@ -0,0 +1,69 @@
+        ctx_rate = context.copy()
+        ctx_rate['date'] = self.revaluation_date
+        ctx_rate['monthly_rate'] = True
+        # ctx_rate['revaluation'] = True
```
left by mistake?

Tonow-c2c reviewed (approved):
LGTM Just code review

grindtildeath committed:
Drop unneeded override of res.currency

The override of res.currency _get_conversion_rate did return 1.0 / from_currency.rate
when 'revaluation' was True in context.
However, as res.currency.rate is defined per company and the context is only
used for a computation to the company currency, the std method from odoo will also
return 1.0 / from_currency.rate, because to_currency is the company currency.
(https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/res/res_currency.py#L179)
Therefore, we do not need this override.

grindtildeath committed:
Remove commented line

This line was already commented in v7.0, so we do not need to keep it across all versions

grindtildeath commented on account_multicurrency_revaluation/model/res_currency.py:
```diff
@@ -14,6 +14,7 @@ class ResCurrency(models.Model):
 
     @api.model
     def _get_conversion_rate(self, from_currency, to_currency):
+        # TODO get rid of this method
```
Please go ahead. cf https://github.com/OCA/account-closing/pull/102/commits/df4e429e47f72f9df507c889d0f5b3d21e3bbca3

grindtildeath reviewed (changes requested):
I guess most of my comments from https://github.com/OCA/account-closing/pull/106#pullrequestreview-221592612 also applies here.

simahawk commented:
lint check :red_circle:

simahawk commented on account_multicurrency_revaluation_rate_type/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+
+{
+    "name": "Multicurrency revaluation with monthly currency rates",
+    "version": "10.0.1.0.1",
```
if you add it now, it should be 1.0.0

simahawk commented on account_multicurrency_revaluation_rate_type/model/res_company.py:
```diff
@@ -0,0 +1,16 @@
+            ('average', 'Average'),
+            ('daily', 'Daily'),
+        ],
+        default='daily')
```
same as #106

simahawk reviewed (approved):
skimmed thru it: LG but - as per #106 - I leave the last word to @grindtildeath :wink:

Commit:
[IMP] cleanup pylint checks
